### PR TITLE
fix: add guard to the controllers

### PR DESCRIPTION
**Pull request by AlejandraRochaOMcKinsey**
## Issue ticket number and link
https://trello.com/c/SHag9ndT/377-implementar-guard-de-roles-en-endpoints-privados

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [ ] :sparkles: Feature
- [x] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

- No role

## What is the new behavior?

- Restric access to user roles

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/modules/events/event.controller.ts [hunk 1]
```diff
@@ -10,21 +10,23 @@ import {
...
 
 @Controller('v2/events')
+@UseGuards(AuthGuard('jwt'), RolesGuard)
 export class EventController {
   constructor(private readonly eventService: EventService, private logger: Logger) {}
 
   @Post()
-  @GuardRol('admin')
-  @UseGuards(RolesGuard)
+  @Roles('admin')
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.ts:29]: if they use string constant, is better centralized in a const file, that could be a environment value too

File: docs/FIREBASE.md [hunk 2]
```diff
@@ -62,5 +62,14 @@ If you prefer to interact with Firebase Authentication through a direct API endp
 }
 ```
 
+### FirebaseAuthStrategy
+
+The FirebaseAuthStrategy is part of our security infrastructure, leveraging JWT for authentication. The secretOrKey parameter is pivotal, ensuring the JWT's integrity by matching it against a known secret.
+
+- Purpose: Validates JWT tokens' signatures to confirm their authenticity and integrity.
+- Configuration: Sourced from the PRIVATE_KEY environment variable. It's critical for the PRIVATE_KEY to remain confidential to maintain security.
+- Implementation: The strategy utilizes this key to decode and validate incoming JWTs in the Authorization header of requests.
```
Comment (FelipeLosada1996) [inline docs/FIREBASE.md:71]: Where can I find the value?

File: src/infrastructure/environment.js [hunk 3]
```diff
@@ -29,6 +29,8 @@ function getEnvironmentVariables() {
     MONGO_PORT: process.env.DB_PORT,
     MONGO_USER: process.env.DB_ROOT_USERNAME,
     MONGO_PASSWORD: process.env.DB_ROOT_PASSWORD,
+    ADMIN_ROLE: process.env.ADMIN_ROLE,
```
Comment (FelipeLosada1996) [inline src/infrastructure/environment.js:32]: you can set by default admin, to not be force to has a first value with a |

File: src/modules/events/event.controller.ts [hunk 4]
```diff
@@ -10,21 +10,25 @@ import {
...
 import { EventService } from './event.service'
 import { UpdateEventDto } from './dto/update-event.dto'
 import { EventResponse } from './interfaces/event-response'
 import { RequestGetAllEventsDto } from './dto/request-get-all-events.dto'
+import { AuthGuard } from '@nestjs/passport'
+import { RolesGuard } from '../guard/roles.guard'
+import { Roles } from '../guard/roles.guard.decorator'
+import getEnvironmentVariables from '../../infrastructure/environment'
 
+const {ADMIN_ROLE} = getEnvironmentVariables()
```
Comment (FelipeLosada1996) [inline src/modules/events/event.controller.ts:24]: Is not necessary to be in a variables files, more in a a const file, to don't call it in that way

File: src/infrastructure/environment.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ function getEnvironmentVariables() {
     MONGO_HOST: process.env.DB_HOST,
     MONGO_PORT: process.env.DB_PORT,
     MONGO_USER: process.env.DB_ROOT_USERNAME,
-    MONGO_PASSWORD: process.env.DB_ROOT_PASSWORD,
+    USER_ROLE: process.env.USER_ROLE,
```
Comment (FelipeLosada1996) [inline src/infrastructure/environment.js:31]: You defined in constant file, is not necessary in environment file any more

File: src/common/constants.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const ADMIN_ROLE = 'admin';
```
Comment (FelipeLosada1996) [inline src/common/constants.ts:1]: could be a `const ADMIN_ROLE = 'admin' | process.env.USER_ADMIN_ROLE` ? or is a bad practice?

**Review by FelipeLosada1996 — APPROVED** [review]

**Review by Sofiag8 — APPROVED** [review]

**Pull request merged**